requests/models.py

```diff
@@ -368,17 +384,19 @@ def prepare_url(self, url, params):
         if not host:
             raise InvalidURL("Invalid URL %r: No host supplied" % url)
 
-        # In general, we want to try IDNA encoding every hostname, as that
-        # allows users to automatically get the correct behaviour. However,
-        # we’re quite strict about IDNA encoding, so certain valid hostnames
+        # In general, we want to try IDNA encoding every hostname if unicode string is
```

Comment:
We may want to consider rewording this comment a bit since it no longer matches the logic below.

Perhaps something like,

>In general, we want to try IDNA encoding the hostname if the string contains
  non-ASCII characters. This allows users to automatically get the correct IDNA 
  behaviour. For strings containing only ASCII characters, we need to also verify
  it doesn't start with a wildcard (*), before allowing the unencoded hostname
  through.